Nodejs/Product/Nodejs/Workspace/PackageJsonTestContainerDiscoverer.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestWindow.Extensibility;
+using Microsoft.VisualStudio.Workspace;
+using Microsoft.VisualStudio.Workspace.Indexing;
+using Microsoft.VisualStudio.Workspace.VSIntegration.Contracts;
+
+namespace Microsoft.NodejsTools.Workspace
+{
+    [Export(typeof(ITestContainerDiscoverer))]
+    public sealed class PackageJsonTestContainerDiscoverer : ITestContainerDiscoverer
+    {
+        private readonly IVsFolderWorkspaceService workspaceService;
+
+        private readonly List<PackageJsonTestContainer> containers = new List<PackageJsonTestContainer>();
+        private readonly object containerLock = new object();
+
+        private IWorkspace currentWorkspace;
```

Comment:
Why store this?  Why not use `this.workspaceService.CurrentWorkspace`?  Does that value change more often than we want?